rtloader/three/three.cpp

```diff
@@ -208,7 +211,7 @@ void Three::GILRelease(rtloader_gilstate_t state)
 {
     if (state == DATADOG_AGENT_RTLOADER_GIL_LOCKED) {
         PyGILState_Release(PyGILState_LOCKED);
-    } else {
+    } else if (state == DATADOG_AGENT_RTLOADER_GIL_UNLOCKED) {
```

Comment:
There is a potential race condition as GILEnsure can return `DATADOG_AGENT_RTLOADER_GIL_LOCKED`, but when running `if (state == DATADOG_AGENT_RTLOADER_GIL_LOCKED)`, `_Py_IsFinalizing()` can be true.

Reply:
According to the doc, we don't care about `_Py_IsFinalizing` for `PyGILState_Release `, and they expect each call to be matched, so I think this is correct.